utils/kubernetes/describe/describe_test.go

```diff
@@ -0,0 +1,124 @@
+	Response     string
+	Client       *meshkitkube.Client
+}
+
+func TestDescribe(t *testing.T) {
+	ctx := context.Background()
+	//set up mock client  to handle api request and return expected responses
+	mckpod := MockClient{
+		Object:       "Pod",
+		Method:       "GET",
+		RequestUrl:   fmt.Sprintf("%s/api/v1/namespaces/default/pods?limit=10&continue=", ctx),
+		ResponseCode: 200,
+		Response:     "Mock response for GET request to /api/v1/namespaces/default/pods?limit=10&continue=abcdefg",
+	}
+	mckDeployment := MockClient{
+		Object:       "Deployment",
+		Method:       "POST",
+		RequestUrl:   fmt.Sprintf("%s/apis/apps/v1/namespaces/default/deployments", ctx),
+		ResponseCode: 200,
+		Response:     "Mock response for POST request to /apis/apps/v1/namespaces/default/deployments",
+	}
+	mckJob := MockClient{
+
+		Object:       "Job",
+		Method:       "GET",
+		RequestUrl:   fmt.Sprintf("%s/api/v1/job", ctx),
+		ResponseCode: 200,
+		Response:     "Moc Response for GET request to /api/v1/namespace",
+	}
+	mckcronjob := MockClient{
+
+		Object:       "Cronjob",
+		Method:       "PUT",
+		RequestUrl:   fmt.Sprintf("%s/apis/batch/v1beta1/namespaces/default/cronjobs", ctx),
+		ResponseCode: 200,
+		Response:     "Mock response for PUT request to /apis/batch/v1beta1/namespaces/default/cronjobs",
+	}
+
+	//create test cases
+	testCases := []struct {
+		Name          string
+		Options       DescriberOptions
+		ExpectedError error
+		clientApi     MockClient
+		Client        MockClient
+	}{
+		{
+			Name: "Pod",
+			Options: DescriberOptions{
+				Name:      "mypod",
+				Namespace: "test",
+				Type:      Pod,
+			},
+			ExpectedError: nil,
+			clientApi:     mckpod,
+		},
+		{
+			Name: "Deployment",
+			Options: DescriberOptions{
+				Name:      "deployment",
+				Namespace: "test",
+				Type:      Deployment,
+			},
+			ExpectedError: nil,
+			clientApi:     mckDeployment,
+		},
+		{
+			Name: "Job",
+			Options: DescriberOptions{
+				Name:      "job",
+				Namespace: "test",
+				Type:      Job,
+			},
+			ExpectedError: nil,
+			clientApi:     mckJob,
+		},
+		{
+			Name: "CronJob",
+			Options: DescriberOptions{
+				Name:      "cronjob",
+				Namespace: "test",
+				Type:      CronJob,
+			},
+			ExpectedError: nil,
+			clientApi:     mckcronjob,
+		},
+	}
+
+	// run test cases
+	for _, tc := range testCases {
+		t.Run(tc.Name, func(t *testing.T) {
+			// Register mock responders for each mock client
+			httpmock.RegisterResponder(tc.clientApi.Method, tc.clientApi.RequestUrl,
+				httpmock.NewStringResponder(tc.clientApi.ResponseCode, tc.clientApi.Response))
+
+			output, err := Describe(tc.Client.Client, tc.Options)
+			//check response for each mockclient
+			if output != tc.clientApi.Response {
+				t.Errorf("Test case %s failed expected %s , but got %s", tc.Name, tc.clientApi.Response, output)
+			}
+			if err != nil {
+				t.Errorf("Testcase failed for %s, coudnt get %s got %s, ", tc.Name, tc.ExpectedError, err)
+			}
+
+		})
+	}
+
```

Comment:
And here remove the `whitespace`

Reply:
sorry, i've just done that

Reply:
I saw the CI still gets stuck on the linter below. Maybe you can check it locally, any of `go test` or `golangci-lint` is ok.

```
ok  	github.com/layer5io/meshkit/utils/kubernetes	0.062s
=== RUN   TestDescribe
=== RUN   TestDescribe/Pod
--- FAIL: TestDescribe (0.00s)
    --- FAIL: TestDescribe/Pod (0.00s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x29d50ea]
```

Reply:
i ran go test -v  locally, it gave me the same error

<code>
=== RUN   TestDescribe
=== RUN   TestDescribe/Pod
--- FAIL: TestDescribe (0.00s)
    --- FAIL: TestDescribe/Pod (0.00s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x0 pc=0x239398a]

goroutine 20 [running]:
testing.tRunner.func1.2({0x2596da0, 0x3aea4e0})
        C:/Program Files/Go/src/testing/testing.go:1526 +0x24e
testing.tRunner.func1()
        C:/Program Files/Go/src/testing/testing.go:1529 +0x39f
panic({0x2596da0, 0x3aea4e0})
        C:/Program Files/Go/src/runtime/panic.go:884 +0x213
github.com/layer5io/meshkit/utils/kubernetes/describe.Describe(0x0, {{0x28360da, 0x5}, {0x283476a, 0x4}, 0x0, 0x0, 0x1})
</code>

Reply:
OK, let me check it.

Reply:
Make sure your parameter is not nil

The panic is happened in here https://github.com/meshery/meshkit/blob/d0a5f69c99877c6a18899bb8d972da5a0b02294b/utils/kubernetes/describe/describe.go#L100

You parameter `&client.RestConfig` is nil and it will cause panic.



```
call describe.DescriberFor(kind, &client.RestConfig)
Unable to evaluate expression: error evaluating "&client.RestConfig" as argument clientConfig in function k8s.io/kubectl/pkg/describe.DescriberFor: client is nil
```